### PR TITLE
ani-cli: cleanup package

### PR DESCRIPTION
**Pull request by HaxSam**
###### Description of changes

A little bit of cleanup also moved the `makeBinPath` into `buildInputs` so that `mkShell` knows the dependencies when used in `inputsFrom`
also added some optional dependencies that can be chosen with `--argstr`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/ani-cli/default.nix [hunk 1]
```diff
@@ -22,16 +30,17 @@ stdenvNoCC.mkDerivation rec {
...
 
   nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ gnugrep gnused curl fzf ffmpeg aria2 ]
+                ++ lib.optional (player == "mpv") mpv
+                ++ lib.optional (player == "vlc") vlc
+                ++ lib.optional (player == "iina") iina
+                ++ lib.optional (chromecast != "false") catt
+                ++ lib.optional (sync != "false") syncplay;
 
   installPhase = ''
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:40]: Please don't remove runHooks. This prevents users from a easy `overrideAttrs`.

File: pkgs/applications/video/ani-cli/default.nix [hunk 2]
```diff
@@ -4,10 +4,18 @@
...
 , fzf
 , mpv
+, vlc
+, iina
+, catt
+, syncplay
+, ffmpeg
 , aria2
+, player ? "mpv"
+, chromecast ? "false"
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:17]: Nix has booleans, so `chromecastSupport ? false` is way better here.

File: pkgs/applications/video/ani-cli/default.nix [hunk 3]
```diff
@@ -22,16 +30,17 @@ stdenvNoCC.mkDerivation rec {
   };
 
   nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ gnugrep gnused curl fzf ffmpeg aria2 ]
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:33]: These are not build inputs.

Comment (HaxSam) [reply pkgs/applications/video/ani-cli/default.nix:33]: Is there a way to get the runtime dependencies when I create a shell for this package for example

```nix
{ pkgs ? import <nixpkgs> {} }:

pkgs.mkShell {
  inputsFrom = [ pkgs.ani-cli ];
}
```

that's the reason why I wanted to put them into the build inputs

Comment (wegank) [reply pkgs/applications/video/ani-cli/default.nix:33]: `mkShell` is not supposed to do that. The best you can do is to rename `buildInputs` to `runtimeDependencies`, then explicitly concat `pkgs.ani-cli.runtimeDependencies` with `[ pkgs.ani-cli ]` in your file.

Comment (HaxSam) [reply pkgs/applications/video/ani-cli/default.nix:33]: Ok thank you ^^

File: pkgs/applications/video/ani-cli/default.nix [hunk 4]
```diff
@@ -4,10 +4,18 @@
...
 , mpv
+, vlc
+, iina
+, catt
+, syncplay
+, ffmpeg
 , aria2
+, player ? "mpv"
+, chromecastSupport ? false
+, sync ? false
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:18]:
```suggestion
, syncSupport ? false
```

File: pkgs/applications/video/ani-cli/default.nix [hunk 5]
```diff
@@ -22,14 +30,20 @@ stdenvNoCC.mkDerivation rec {
   };
 
   nativeBuildInputs = [ makeWrapper ];
+  runtimeDependencies = [ gnugrep gnused curl fzf ffmpeg aria2 ]
+    ++ lib.optional (player == "mpv") mpv
+    ++ lib.optional (player == "vlc") vlc
+    ++ lib.optional (player == "iina") iina
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:36]: Probably an assertion before `stdenvNoCC.mkDerivation` that `player` is one of the three.

**Comment by wegank** [comment]
Also, please squash the commits.

File: pkgs/applications/video/ani-cli/default.nix [hunk 6]
```diff
@@ -15,9 +15,13 @@
 , aria2
 , player ? "mpv"
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:16]:
```suggestion
, player ? mpv
```

File: pkgs/applications/video/ani-cli/default.nix [hunk 7]
```diff
@@ -15,9 +15,13 @@
...
 , chromecastSupport ? false
-, sync ? false
+, syncSupport ? false
 }:
 
+let player = lib.optional (player == "mpv") mpv
+  ++ lib.optional (player == "vlc") vlc
+  ++ lib.optional (player == "iina") iina
+in
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:24]:
```suggestion
assert builtins.elem player [ mpv vlc iina ];
```

File: pkgs/applications/video/ani-cli/default.nix [hunk 8]
```diff
@@ -4,12 +4,24 @@
...
+, curl
 , fzf
 , mpv
+, vlc
+, iina
+, catt
+, syncplay
+, ffmpeg
 , aria2
+, player ? "mpv"
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:16]: Please follow the `withMpv` schemes in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/soundkonverter/default.nix.

File: pkgs/applications/video/ani-cli/default.nix [hunk 9]
```diff
@@ -4,12 +4,24 @@
...
+, chromecastSupport ? false
+, syncSupport ? false
 }:
 
+let player = lib.optional (player == "mpv") mpv
+  ++ lib.optional (player == "vlc") vlc
+  ++ lib.optional (player == "iina") iina
+in
```
Comment (wegank) [inline pkgs/applications/video/ani-cli/default.nix:24]:
```suggestion
assert withMpv || withVlc || withIina;
```

**Comment by wegank** [comment]
Again, please squash the commits, thanks.

File: pkgs/applications/video/ani-cli/default.nix (hunk 8 above)
Comment (HaxSam) [reply pkgs/applications/video/ani-cli/default.nix:16]: I looked it up and saw that they also but the runtime deps `buildInputs` I thought you are not supposed to do that?

Comment (wegank) [reply pkgs/applications/video/ani-cli/default.nix:16]: No. I guess it's up to them to clean up the package.

**Review by skykanin — CHANGES_REQUESTED** [review]
Looks fine other than @wegank's comments.

**Comment by wegank** [comment]
Closed by #248966.

**Pull request closed without merging**